Assets/Scripts/ssjAug2018/World/Mailbox/MailboxManager.cs

```diff
@@ -109,9 +107,19 @@ public void ActivateMailboxGroup()
                 letterCount = (letterCount > setSize) ? setSize : letterCount;
                 
                 box.ActivateMailbox(letterCount);
+                _activeMailboxes++;
                 validBoxes.Remove(box);
                 setSize =- letterCount;
             }
         }
+
+        public void MailboxCompleted()
+        {
+            _activeMailboxes--;
+            if(_activeMailboxes <= 0)
+            {
+                ActivateMailboxGroup(FindObjectOfType<Player>());
```

Comment:
FindObjectOfType<>() shouldn't be used anywhere, so definitely need to stop doing that here

Reply:
reason being - it's super, super slow and inefficient

Reply:
Cool. That was a placeholder anyway until we get a better way to fetch the player.